src/test/java/io/confluent/copycat/hdfs/hive/HiveExecTest.java

```diff
@@ -0,0 +1,164 @@
+import org.apache.kafka.copycat.sink.SinkRecord;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.util.HashSet;
+import java.util.Set;
+
+import io.confluent.copycat.avro.AvroData;
+import io.confluent.copycat.hdfs.FileUtils;
+import io.confluent.copycat.hdfs.HdfsSinkConnectorTestBase;
+import io.confluent.copycat.hdfs.HdfsWriter;
+
+import static org.junit.Assert.assertEquals;
+
+public class HiveExecTest extends HdfsSinkConnectorTestBase {
+
+  @Test
+  public void testAvroTableImport() throws Exception {
+    String topic = "test";
+    Schema avroSchema = AvroData.fromCopycatSchema(createSchema());
+    HiveExec hiveExec = new HiveExec(conf, connectorConfig);
+    runHive(hiveExec, "DROP TABLE " + topic);
+    hiveExec.createAvroTable(topic, avroSchema);
+
+    String tablesInHive = runHive(hiveExec, "SHOW TABLES");
+    assertEquals(tablesInHive, topic);
+
+    String description = runHive(hiveExec, "DESCRIBE " + topic);
+    String[] colTypes = {"boolean", "int", "bigint", "float", "double"};
+    String[] lines = description.split("\n");
+    for (int i = 0; i < colTypes.length; ++i) {
+      String[] parts = lines[i].replace(" ", "").split("\t");
+      assertEquals(colTypes[i], parts[1]);
+    }
+  }
+
+  @Test
+  public void testAddPartition() throws Exception {
+    String topic = "test";
+    int partition = 0;
+    TopicPartition topicPart = new TopicPartition(topic, partition);
+
+    Set<TopicPartition> assignment = new HashSet<>();
+    assignment.add(topicPart);
+
+    HdfsWriter hdfsWriter = new HdfsWriter(connectorConfig, assignment);
+    hdfsWriter.recover(topicPart);
+
+    String key = "key";
+    org.apache.kafka.copycat.data.Schema schema = createSchema();
+    Struct record = createRecord(schema);
+
+    for (long offset = 0; offset < 7; offset++) {
+      SinkRecord sinkRecord =
+          new SinkRecord(topic, partition, org.apache.kafka.copycat.data.Schema.STRING_SCHEMA, key,
+                         schema, record, offset);
+      hdfsWriter.writeRecord(topicPart, sinkRecord);
+    }
+    hdfsWriter.close();
+    // TODO: Move WALs to proper directories
+    fs.delete(new Path(FileUtils.logFileName(url, topicsDir, topicPart)), false);
+
+    Schema avroSchema = AvroData.fromCopycatSchema(createSchema());
+    HiveExec hiveExec = new HiveExec(conf, connectorConfig);
+
+    runHive(hiveExec, "DROP TABLE " + topic);
+
+    hiveExec.createAvroTable(topic, avroSchema);
+    hiveExec.addPartition(topic, 0);
+    hiveExec.setPartitionLoc(topic, 0);
+
+    String[] expectedResult = {"true", "12", "12", "12.2", "12.2", "0"};
+    String result = runHive(hiveExec, "SELECT * from " + topic);
+    String[] rows = result.split("\n");
+    assertEquals(7, rows.length);
+    for (String row: rows) {
+      String[] parts  = row.replace(" ", "").split("\t");
```

Comment:
If we keep doing this, we might just want to refactor it into a method. It's not confusing or anything, but just for clarity.

Also, will this work if any of the returned items are strings?